src/main/java/seedu/weme/model/UserPrefs.java

```diff
@@ -70,6 +80,15 @@ public void setMemeImagePath(Path memeImagePath) {
         this.memeImagePath = memeImagePath;
     }
 
+    public ObservableMap<String, String> getObservableUserSettings() {
```

Comment:
two things regarding this,
first, I don't think this is going to work when you set values, as it modifies the instance field Strings and not this map, you will likely have to revamp the whole UserPrefs to make it observable, and make the GUI listen to it.
second, I think it would be nice to see the full file path for the data, meme and template, not sure how this should be reflected in this class file though. Ideally, it should be as it is right now in this class file, but in the GUI, it should appear as the full path.

Reply:
Will take note if I were to make the HashMap mutable. And I will also consider setting everything to absolute path (if that is what you mean) for better consistency. Thanks for the input.

Reply:
to clarify, i meant absolute path just for what is shown the to user in the preferences tab.

Reply:
Yeah I know